### PR TITLE
freesurfer_dir in `data-config build` tool

### PR DESCRIPTION
**Pull request by amygutierrez**
## Fixes
Adds `freesurfer_dir` in data_settings, so that you can add the freesurfer directory for corresponding participants when using `data-config build` tool.
Fixes #[issue number] by @[original issue author]

## Description
When using `data-config build` tool in C-PAC, there is no way to add `freesurfer_dir` paths to participants. 

## Technical details
Added `freesurfer_dir` to data_settings.yml. 
Added freesurfer_dir into multiple function blocks so that it can parse the participant list and add them to its corresponding location.

## Tests

Tested by running tool `data-config build` on subjects with freesurfer directories and 2 functional runs.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/utils/build_data_config.py [hunk 1]
```diff
@@ -1268,15 +1293,19 @@ def update_data_dct(file_path, file_template, data_dct=None, data_type="anat",
 
         data_dct[site_id][sub_id][ses_id]['brain_mask'] = file_path
 
-    elif data_type == "func":
-
+    elif 'func' in data_type:
```
Comment (shnizzedy) [inline CPAC/utils/build_data_config.py:1296]: Why change https://github.com/FCP-INDI/C-PAC/blob/50f39a28a77351563076dcb9636139ceb247e6c7/CPAC/utils/build_data_config.py#L1251 https://github.com/FCP-INDI/C-PAC/blob/50f39a28a77351563076dcb9636139ceb247e6c7/CPAC/utils/build_data_config.py#L1271 to https://github.com/FCP-INDI/C-PAC/blob/a693b608dde98fe9ea28a6bfab5e365710e5c60d/CPAC/utils/build_data_config.py#L1276 https://github.com/FCP-INDI/C-PAC/blob/a693b608dde98fe9ea28a6bfab5e365710e5c60d/CPAC/utils/build_data_config.py#L1296 ? It's fine, just curious

File: CPAC/utils/build_data_config.py [hunk 2]
```diff
@@ -1268,15 +1293,19 @@ def update_data_dct(file_path, file_template, data_dct=None, data_type="anat",
...
 
         # scan parameters time
         scan_params = None
         if scan_params_dct:
             scan_params = find_unique_scan_params(scan_params_dct, site_id,
                                                   sub_id, ses_id, scan_id)
+        for file in contents:
+            if '.json' in file:
+                scan_params = os.path.join(os.path.dirname(file_path), file)
+
```
Comment (shnizzedy) [inline CPAC/utils/build_data_config.py:1308]: Does https://github.com/FCP-INDI/C-PAC/blob/a693b608dde98fe9ea28a6bfab5e365710e5c60d/CPAC/utils/build_data_config.py#L1298-L1307 correctly match up the params with the scans when there're multiple scans?

File: CPAC/utils/build_data_config.py (hunk 1 above)
Comment (amygutierrez) [reply CPAC/utils/build_data_config.py:1296]: i just like how that looks more than "==", but can change back

File: CPAC/utils/build_data_config.py (hunk 2 above)
Comment (amygutierrez) [reply CPAC/utils/build_data_config.py:1308]: Yes, when there are 2 functional scans, the snac parameters match. This is a screen shot provided by @nathaliaesper when she tested out this branch 
image

Comment (amygutierrez) [reply CPAC/utils/build_data_config.py:1308]: Just spoke with Nathalia on some tests she did on this branch, and she noticed that its using one scan parameters for both scans. So I'll fix that and ask for a review again 
image

File: CPAC/utils/build_data_config.py (hunk 1 above)
Comment (shnizzedy) [reply CPAC/utils/build_data_config.py:1296]: no need, like I said, just curious

**Comment by amygutierrez** [comment]
The errors above have been fixed and have been tested by @nathaliaesper

**Review by shnizzedy — APPROVED** [review]

**Pull request merged**